libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts

```diff
@@ -0,0 +1,231 @@
+            break;
+          }
+          case CohortColumnNames.TrueY: {
+            const filter = translatePreBuiltCohortFilterForTarget(
+              preBuiltCohortFilter,
+              jointDataset,
+              CohortColumnNames.TrueY
+            );
+            filterList.push(filter);
+            break;
+          }
+          case CohortColumnNames.ClassificationOutcome: {
+            const filter =
+              translatePreBuiltCohortFilterForClassificationOutcome(
+                preBuiltCohortFilter,
+                jointDataset
+              );
+            filterList.push(filter);
+            break;
+          }
+          case CohortColumnNames.Index: {
+            const filter: IFilter = {
+              arg: preBuiltCohortFilter.arg,
+              column: JointDataset.IndexLabel,
+              method: preBuiltCohortFilter.method
+            } as IFilter;
+            filterList.push(filter);
+            break;
+          }
+          case CohortColumnNames.RegressionError: {
+            const filter: IFilter = {
+              arg: preBuiltCohortFilter.arg,
+              column: JointDataset.RegressionError,
+              method: preBuiltCohortFilter.method
+            } as IFilter;
+            filterList.push(filter);
+            break;
+          }
+          default: {
+            const [filter, errorString] =
+              translatePreBuiltCohortFilterForDataset(
+                preBuiltCohortFilter,
+                jointDataset
+              );
+            if (filter !== undefined) {
+              filterList.push(filter);
+            } else if (errorString !== undefined) {
+              errorStrings.push(errorString);
+            }
+
+            break;
+          }
+        }
+      }
+      const errorCohortEntry = new ErrorCohort(
+        new Cohort(preBuiltCohort.name, jointDataset, filterList),
+        jointDataset
+      );
+      errorCohortList.push(errorCohortEntry);
+    }
+  }
+  return [errorCohortList, errorStrings];
+}
+
+function translatePreBuiltCohortFilterForTarget(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset,
+  cohortColumnName: CohortColumnNames
+): IFilter {
+  let filterColumnName = JointDataset.PredictedYLabel;
+  if (cohortColumnName === CohortColumnNames.TrueY) {
+    filterColumnName = JointDataset.TrueYLabel;
+  }
+  if (
+    jointDataset.getModelType() === ModelTypes.Multiclass ||
+    jointDataset.getModelType() === ModelTypes.Binary
+  ) {
+    const modelClasses = jointDataset.getModelClasses();
+    const index: number[] = [];
+    for (const modelClass of preBuiltCohortFilter.arg) {
+      const indexModelClass = modelClasses.indexOf(modelClass);
+
+      if (indexModelClass !== -1) {
+        index.push(indexModelClass);
+      }
+    }
+
+    index.sort((a, b) => a - b);
+
+    const filter: IFilter = {
+      arg: index,
+      column: filterColumnName,
+      method: preBuiltCohortFilter.method
+    } as IFilter;
```

Comment:
is there any reason we do type casting?
Typescript isn't a very strong on type checking, so this type casting can generate run time issue.

Reply:
If it is due to missing values, assign some default maybe better.

Reply:
I don't know how to initialize an object from an interface in typescript. The other examples in the code that create a IFilter object do this in this way so copied the pattern. https://github.com/microsoft/responsible-ai-toolbox/blob/f6840f1424992dab2041ee13f040734e9ac6ada1/libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortEditor.tsx#L450

Is there another way to do this?

Reply:
It does not need any extra step, unless IFilter contains some extra required property.
Your assignment should work without type casting.
const filter: IFilter = {
column: xxx
arg:index,
method: xxx
}